File: src/main/scala/com/bitdecay/tictactoe/scala/condition/TieCondition.scala
package com.bitdecay.tictactoe.scala.condition

import com.bitdecay.board.{Action, Condition, GameBoardState}
import com.bitdecay.tictactoe.scala.State
import com.bitdecay.tictactoe.scala.action.TieAction

object TieCondition extends Condition[State, TieAction](new TieAction) {

  override def check(prev: State, cur: State): Boolean = !prev.finished && !cur.finished && cur.q != 0 && cur.w != 0 && cur.e != 0 && cur.a != 0 && cur.s != 0 && cur.d != 0 && cur.z != 0 && cur.x != 0 && cur.c != 0

  override def description(): String = ???

  override def serialize(): String = ???

  override def deserialize(s: String) = ???
}

File: src/main/java/com/bitdecay/tictactoe/condition/TieCondition.java
package com.bitdecay.tictactoe.condition;

import com.bitdecay.board.Condition;
import com.bitdecay.tictactoe.State;
import com.bitdecay.tictactoe.action.TieAction;

public class TieCondition extends Condition<State, TieAction> {
    public TieCondition() {
        super(new TieAction());
    }

    @Override
    public boolean check(State previous, State current) {
        return !previous.finished && !current.finished &&
                current.q != 0 &&
                current.w != 0 &&
                current.e != 0 &&
                current.a != 0 &&
                current.s != 0 &&
                current.d != 0 &&
                current.z != 0 &&
                current.x != 0 &&
                current.c != 0;
    }

    @Override
    public String description() {
        return null;
    }

    @Override
    public String serialize() {
        return null;
    }

    @Override
    public Condition deserialize(String s) {
        return null;
    }
}
